manifests/profile/networking/private.pp

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2018 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+
+# nebula::profile::networking::private
+#
+# Configure private network interface. The private address is constructed by
+# replacing the last octet of address_template with the last octet of the
+# public IP address.
+#
+# @example
+#   include nebula::profile::networking::private
+class nebula::profile::networking::private (
+  String $address_template = '192.168.0.%s',
+  String $netmask = '255.255.255.0',
+  String $network = '192.168.0.0',
+  String $broadcast = '192.168.255.255',
+  Optional[String] $interface = undef,
+) {
+
+  if !$interface and $facts['os']['release']['major'] == '9' and $facts['is_virtual']
+    and $facts['networking']['interfaces'].has_key('ens4') {
```

Comment:
```suggestion
    and has_key($facts['networking']['interfaces'], 'ens4') {
```

Puppet hashes don't have methods—gotta go old-school C-style objects on this one.